### ISSUE TITLE
Mac Catalyst support for match

### ISSUE THREAD

**Issue by molind** [fastlane/fastlane#15514]
#### Motivation Behind Feature
It's easy to convert iOS app to macOS app using Catalyst. And it would be really helpful to be able to enable Catalyst support in the match. 

#### Feature Description

Detect when an app identifier has enabled mac support. And generate a macOS provisioning profile for that app identifier. Or if it's hard to detect that app identifier has mac support. Add this parameter to the `Matchfile`.
`match development` should generate a provisioning profile for iOS and macOS.

#### Alternatives or Workarounds

Right now I create a provisioning profile manually and share it with my team. Development certs and iOS provisioning profiles are managed by match. Only the macOS provisioning profile is shared manually.

**Comment by SpectralDragon** [comment]
I got a same problem, I can't generate catalyst version.

**Comment by kylekurz** [comment]
@fastlane-bot this is still an issue and should remain open/active.

**Comment by crewshin** [comment]
+1

**Comment by joshdholtz** [comment]
Will be getting to this soon! 😊

**Comment by michallaskowski** [comment]
Hi. We made kind of a hack to access Developer Id certificate with Fastlane Match. I used our enterprise branch to import a Developer Id certificate as a 'distribution' one, manually via ruby and git operations. Creating distribution certs seems anyway impossible on Apple Dev portal for enterprise accounts, so we should be ok. And our developer id cert is from enterprise account too.

This way we can use 
```
sync_code_signing(
      type: "appstore", # developer id cert is under cert/distribution folder
      git_branch: "enterprise",
      skip_provisioning_profiles: true,
      app_identifier: "some.bundle.id.here" # profile will be skipped anyway
      )
```
to install the profile on our CI machines.

And also we will store right now provisioning profile used for notarization in the repo with the project, and will install it manually before building our project.

I would like to know what will be the scope of the work, and what will be the setup after. Will there be a new `type` for sync_code_signing, and would it support Developer Id certificate? Then also will there be a support for a provisioning profile used for notarization?
Thanks

**Comment by triplef** [comment]
It looks like this was recently done: #15918.

**Comment by molind** [comment]
@triplef Yes, finally! :) Thanks everyone.

**Comment by chrisballinger** [comment]
@joshdholtz I'm not sure if this was actually fixed, or there may have been a regression. It seems like `match` still doesn't generate Mac Catalyst App Store profiles when using Xcode 11.4 and `DERIVE_MACCATALYST_PRODUCT_BUNDLE_IDENTIFIER = NO` (for Universal Purchase). I don't have any App IDs listed that have the `maccatalyst` prefix, so maybe the old logic in #15918 to detect that will no longer work. 

```
error: exportArchive: No "Mac Catalyst App Store" profiles for team 'Chris Ballinger' matching 'match AppStore com.example.Example' are installed.
```

I was able to manually generate one on the developer portal and import it via `match import`, but match detects it as a "Distribution" certificate, not as an App Store cert.

**Comment by joshdholtz** [comment]
@chrisballinger I got a PR out for this I’m working on finishing up that fixes this! Its in #16158 😇 Had to completely rewrite all API interactions that sigh/match did with provisioning profiles to use the new App Store Connect because the old legacy API doesn’t support the new Catalyst profile types.

If you want to try using that branch that would actually be 💯 I’ve used in on a few different projects and it looks like it works pretty good but I need to do a lot of work on the tests yet to make that PR mergable.
